src/Swoole/Request.php

```diff
@@ -148,7 +148,16 @@ public function getProtocol(): string
             return 'http';
         }
 
-        return $protocol;
+        switch ($protocol) {
+            case 'http':
+            case 'https':
+            case 'ws':
+            case 'wss':    
+                return $protocol;
+                break;
+            default:
+                return 'https';
+        }
```

Comment:
```suggestion
        return match ($protocol) {
            'http', 'https', 'ws', 'wss' => $protocol,
            default => 'https'
        };
```

Using match might be the faster solution, since this is done on every request 👍🏻